xls/fuzzer/value_generator.cc

```diff
@@ -193,20 +195,40 @@ absl::StatusOr<Expr*> GenerateDslxConstant(absl::BitGenRef bit_gen,
                                            Module* module,
                                            TypeAnnotation* type) {
   dslx::Span fake_span = dslx::FakeSpan();
-  if (auto* builtin_type = dynamic_cast<dslx::BuiltinTypeAnnotation*>(type);
-      builtin_type != nullptr) {
+
+  if (std::optional<dslx::BitVectorMetadata> metadata =
+          dslx::ExtractBitVectorMetadata(type);
+      metadata.has_value()) {
+    auto bit_count_or = absl::visit(
```

Comment:
maybe a silly nit - but couldn't this just be `XLS_ASSIGN_OR_RETURN(int64_t bit_count, absl::visit(...))`?

Reply:
I was wondering a similar thing (due to comment above), though I could see the argument for reducing nesting for readability.

Reply:
I didn't want to put a large and complex expression inside the macro. If you feel strongly I'll change it.